### PR TITLE
Enable profile section for OIDC and SAML sample applications

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## Purpose
This PR solves the issue reported through #144 

## Goals
Goal was to enable profile section for SAML and OIDC samples.

## Approach
QSG script changes were done to enable claim mappings. Then, claims were extracted from SAML and OIDC responses.

For claim extraction, Claim management APIs are used. This enables to extract correct display values of each claim

## User stories
#144

## Release note
NA

## Documentation
We can add profile section related details to qsg documentation

## Training
NA

## Certification
NA

## Marketing
NA

## Automation tests
NA

## Security checks
NA

## Samples
NA

## Related PRs
#143 should be accepted before this

## Migrations (if applicable)
NA

## Test environment
NA
 
## Learning
NA

### CODE REVIEW

File: quick-start-guide/src/main/resources/bin/qsg.sh [hunk 1]
```diff
@@ -1864,6 +1841,38 @@ echo "<soapenv:Envelope xmlns:soapenv="\"http://schemas.xmlsoap.org/soap/envelop
                     <xsd1:alwaysSendMappedLocalSubjectId>false</xsd1:alwaysSendMappedLocalSubjectId>
                     <!--Optional:-->
                     <xsd1:localClaimDialect>true</xsd1:localClaimDialect>
+                     <xsd1:claimMappings>
```
Comment (Kavindu-Dodan) [inline quick-start-guide/src/main/resources/bin/qsg.sh]: @omindu  Here is the SOAP message 

<<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://org.apache.axis2/xsd" xmlns:xsd1="http://model.common.application.identity.carbon.wso2.org/xsd">
    <soapenv:Header/>
    <soapenv:Body>
        <xsd:updateApplication>
            
            <xsd:serviceProvider>
                
                <xsd1:applicationID>354</xsd1:applicationID>
                
                <xsd1:applicationName>dispatch</xsd1:applicationName>
                
                <xsd1:claimConfig>
                    
                    <xsd1:alwaysSendMappedLocalSubjectId>false</xsd1:alwaysSendMappedLocalSubjectId>
                    
                    <xsd1:localClaimDialect>true</xsd1:localClaimDialect>
                    <xsd1:claimMappings>
	                  
	                  <xsd1:localClaim>
	                     
	                     <xsd1:claimUri>http://wso2.org/claims/fullname</xsd1:claimUri>
	                  </xsd1:localClaim>
	                  
	                  <xsd1:mandatory>true</xsd1:mandatory>
	                  
	                  <xsd1:remoteClaim>
	                     
	                     <xsd1:claimUri>http://wso2.org/claims/fullname</xsd1:claimUri>
	                  </xsd1:remoteClaim>
	                  
	                  <xsd1:requested>true</xsd1:requested>
            		</xsd1:claimMappings>
            		
            		<xsd1:claimMappings>
	                  
	                  <xsd1:localClaim>
	                     
	                     <xsd1:claimUri>http://wso2.org/claims/emailaddress</xsd1:claimUri>
	                  </xsd1:localClaim>
	                  
	                  <xsd1:mandatory>true</xsd1:mandatory>
	                  
	                  <xsd1:remoteClaim>
	                     
	                     <xsd1:claimUri>http://wso2.org/claims/emailaddress</xsd1:claimUri>
	                  </xsd1:remoteClaim>
	                  
	                  <xsd1:requested>true</xsd1:requested>
            		</xsd1:claimMappings>

                </xsd1:claimConfig>
                
                <xsd1:description>sample service provider</xsd1:description>
                
                <xsd1:inboundAuthenticationConfig>
                    
                    <xsd1:inboundAuthenticationRequestConfigs>
                        
                        <xsd1:inboundAuthKey>saml2-web-app-pickup-dispatch.com</xsd1:inboundAuthKey>
                        
                        <xsd1:inboundAuthType>samlsso</xsd1:inboundAuthType>
                        
                        <xsd1:properties>
                            
                            <xsd1:name>attrConsumServiceIndex</xsd1:name>
                            
                            <xsd1:value>1223160755</xsd1:value>
                        </xsd1:properties>
                    </xsd1:inboundAuthenticationRequestConfigs>
                </xsd1:inboundAuthenticationConfig>
                
                <xsd1:inboundProvisioningConfig>
                    
                    <xsd1:provisioningEnabled>false</xsd1:provisioningEnabled>
                    
                    <xsd1:provisioningUserStore>PRIMARY</xsd1:provisioningUserStore>
                </xsd1:inboundProvisioningConfig>
                
                <xsd1:localAndOutBoundAuthenticationConfig>
                    
                    <xsd1:alwaysSendBackAuthenticatedListOfIdPs>false</xsd1:alwaysSendBackAuthenticatedListOfIdPs>
                    
                    <xsd1:authenticationStepForAttributes></xsd1:authenticationStepForAttributes>
                    
                    <xsd1:authenticationStepForSubject></xsd1:authenticationStepForSubject>
                    <xsd1:authenticationType>default</xsd1:authenticationType>
                    
                    <xsd1:subjectClaimUri>http://wso2.org/claims/fullname</xsd1:subjectClaimUri>
                </xsd1:localAndOutBoundAuthenticationConfig>
                
                <xsd1:outboundProvisioningConfig>
                    
                    <xsd1:provisionByRoleList></xsd1:provisionByRoleList>
                </xsd1:outboundProvisioningConfig>
                
                <xsd1:permissionAndRoleConfig></xsd1:permissionAndRoleConfig>
                
                <xsd1:saasApp>false</xsd1:saasApp>
            </xsd:serviceProvider>
        </xsd:updateApplication>
    </soapenv:Body>
</soapenv:Envelope>

**Comment by Kavindu-Dodan** [comment]
@omindu  Regarding SOAP messages, you can view samples SOAP messages that are bundled with the scenarios. When QSG scripts run, they get updated with correct values (ex:- with SP name, IDP port etc.)

Please check below links,

https://github.com/wso2/samples-is/blob/master/quick-start-guide/src/main/resources/QSG/bin/02/sso-config-dispatch.xml

https://github.com/wso2/samples-is/blob/master/quick-start-guide/src/main/resources/QSG/bin/02/update-app-dispatch.xml

File: sso-samples/oidc-sso-sample/pickup-dispatch/src/main/webapp/home.jsp [hunk 2]
```diff
@@ -225,48 +242,25 @@
...
-                                        <td><%=key%>
-                                        </td>
-                                        <td><%=hashmap.get(key).toString()%>
-                                        </td>
-                                    </tr>
-                                    <%
-                                            }
-                                        }
-                                    %>
+                                        <%for(String claim:oidcClaimDisplayValueMap.keySet()){%>
```
Comment (omindu) [inline sso-samples/oidc-sso-sample/pickup-dispatch/src/main/webapp/home.jsp]: Fix formatting

File: etc/claim-manager/pom.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (Kavindu-Dodan) [reply etc/claim-manager/pom.xml]: Adde in the latest commit

File: sso-samples/oidc-sso-sample/pickup-dispatch/src/main/webapp/home.jsp (hunk 2 above)
Comment (Kavindu-Dodan) [reply sso-samples/oidc-sso-sample/pickup-dispatch/src/main/webapp/home.jsp]: Fixed in the latest commit

**Comment by Kavindu-Dodan** [comment]
@omindu can we accept this pr :)

File: sso-samples/saml2-sso-sample/saml2-web-app-pickup-manager/src/main/webapp/home.jsp [hunk 4]
```diff
@@ -379,39 +393,29 @@
...
-
+                                <% if (!subjectAttributeValueMap.isEmpty()) { %>
+                                    <table class="table">
+                                        <thead>
+                                        <tr>
+                                            <th rowspan="2">User Details</th>
+                                        </tr>
+                                        </thead>
+                                        <tbody>
+                                            <%for(String attribute:subjectAttributeValueMap.keySet()){%>
```
Comment (omindu) [inline sso-samples/saml2-sso-sample/saml2-web-app-pickup-manager/src/main/webapp/home.jsp]: Fix formatting

File: sso-samples/saml2-sso-sample/saml2-web-app-pickup-manager/src/main/webapp/home.jsp [hunk 5]
```diff
@@ -379,39 +393,29 @@
...
-                                    <tr>
-                                        <p align="center"> Please configure claim mapping in SP configurations.</p>
-                                    </tr>
-                                    <%
-                                        }
-                                    %>
-                                    </tbody>
-                                </table>
-
+                                <% if (!subjectAttributeValueMap.isEmpty()) { %>
```
Comment (omindu) [inline sso-samples/saml2-sso-sample/saml2-web-app-pickup-manager/src/main/webapp/home.jsp]: Is there a possibility for `subjectAttributeValueMap` to be null ? If so add a null check

File: sso-samples/oidc-sso-sample/pickup-dispatch/src/main/webapp/home.jsp [hunk 6]
```diff
@@ -225,48 +242,25 @@
...
-                                        <td><%=key%>
-                                        </td>
-                                        <td><%=hashmap.get(key).toString()%>
-                                        </td>
-                                    </tr>
-                                    <%
-                                            }
-                                        }
-                                    %>
+                                        <% for(String claim:oidcClaimDisplayValueMap.keySet()){ %>
```
Comment (omindu) [inline sso-samples/oidc-sso-sample/pickup-dispatch/src/main/webapp/home.jsp]: Formatting

File: sso-samples/oidc-sso-sample/pickup-manager/src/main/webapp/home.jsp [hunk 7]
```diff
@@ -385,46 +398,25 @@
...
-                                    %>
-                                    <tr>
-                                        <td><%=key%></td>
-                                        <td><%=hashmap.get(key).toString()%></td>
-                                    </tr>
-                                    <%
-                                            }
-                                        }
-                                    %>
+                                        <% for(String claim:oidcClaimDisplayValueMap.keySet()){ %>
```
Comment (omindu) [inline sso-samples/oidc-sso-sample/pickup-manager/src/main/webapp/home.jsp]: formatting

File: sso-samples/oidc-sso-sample/pickup-dispatch/src/main/webapp/home.jsp [hunk 8]
```diff
@@ -46,13 +49,23 @@
...
             name = SignedJWT.parse(idToken).getJWTClaimsSet().getSubject();
-            claimsSet = SignedJWT.parse(idToken).getJWTClaimsSet();
+            ReadOnlyJWTClaimsSet claimsSet = SignedJWT.parse(idToken).getJWTClaimsSet();
+            
+            ClaimManagerProxy claimManagerProxy = (ClaimManagerProxy) application.getAttribute("claimManagerProxyInstance");
+
+            customClaimValueMap = claimsSet.getCustomClaims();
+            
+            oidcClaimDisplayValueMap =
+                    claimManagerProxy.getOidcClaimDisplayNameMapping(new ArrayList<>(customClaimValueMap.keySet()));
```
Comment (omindu) [inline sso-samples/oidc-sso-sample/pickup-dispatch/src/main/webapp/home.jsp]: Better to do a null check before invoking `getOidcClaimDisplayNameMapping`. If the claimManagerProxy is null or retrieveing the display names throws an error, we should still display the claims with from the JWT

File: sso-samples/oidc-sso-sample/pickup-manager/src/main/webapp/home.jsp (hunk 7 above)
Comment (Kavindu-Dodan) [reply sso-samples/oidc-sso-sample/pickup-manager/src/main/webapp/home.jsp]: This is fixed in the latest changes

File: sso-samples/oidc-sso-sample/pickup-dispatch/src/main/webapp/home.jsp (hunk 8 above)
Comment (omindu) [inline sso-samples/oidc-sso-sample/pickup-dispatch/src/main/webapp/home.jsp]: In a case of claim manager proxy not being able to fetch claim mappings, what would be the behavior ?

**Pull request merged**